drivers/session_mock.go

```diff
@@ -2,17 +2,33 @@ package drivers
 
 import (
 	"context"
+	"time"
 
 	"github.com/livepeer/go-livepeer/net"
 	"github.com/stretchr/testify/mock"
 )
 
 type MockOSSession struct {
 	mock.Mock
+	waitForCh bool
+	waitCh    chan struct{}
+	back      chan struct{}
 }
 
-func (s *MockOSSession) SaveData(name string, data []byte, meta map[string]string) (string, error) {
-	args := s.Called()
+func NewMockOSSession() *MockOSSession {
+	return &MockOSSession{
+		waitCh: make(chan struct{}),
```

Comment:
Re: the previous comment on using struct{} vs. interface{} for the channel.

I believe struct{} is typically used for signal channels like these because its the smallest available data type (at least according to https://stackoverflow.com/questions/52035390/why-using-chan-struct-when-wait-something-done-not-chan-interface/52035455). I feel like its probably not a big deal to use one or the other in most cases, but since we mainly use struct{} for signal channels elsewhere in the codebase already that's typically been my default.

Reply:
Thanks! Didn't know that `The empty struct struct {} requires no memory`